R/zchunk_L115.nh3_an_R_S_T_Y.R

```diff
@@ -31,47 +29,112 @@ module_emissions_L115.nh3_an_R_S_T_Y_DISABLED <- function(command, ...) {
+      L115.nh3_tg_R_an_C_Yh.mlt
+
+    # Compute EDGAR emissions by region and sector
+    L115.EDGAR <- EDGAR_NH3
+
+    # Renaming L115.EDGAR variable names - Note: may remove if names for L115.EDGAR change
+    names(L115.EDGAR)[1] <- "IPCC_Annex"
+    names(L115.EDGAR)[2] <- "World_Region"
+
+    # Add gas name and match by emissions factors
+    L115.EDGAR %>%
+      mutate(Non.CO2 = "NH3_AGR") %>%
+      left_join(EDGAR_sector, by = "IPCC") %>%
+      rename(EDGAR_agg_sector = agg_sector) ->
+      L115.EDGAR
+
+    # Convert from EDGAR iso to GCAM_region_ID
+    L115.EDGAR %>%
+      mutate( iso = tolower( ISO_A3 ), ISO_A3 = NULL ) %>%
+      change_iso_code('rou', 'rom') %>%
+      left_join(iso_GCAM_regID, by = "iso") ->
+      L115.EDGAR
+
+    # Note : the old data system drops values from 1970, it is unclear if that is intentional or an accident.
+    # Drop unnecessary columns, make long, and aggregate by region.
+    L115.EDGAR %>%
+      select(-c(IPCC_Annex, World_Region, iso, Name, IPCC, IPCC_description.x, IPCC_description.y, country_name, region_GCAM3)) %>%
+      na.omit() %>%
+      gather(year, value, -c(Non.CO2, EDGAR_agg_sector, GCAM_region_ID)) %>%
+      group_by(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year) %>%
+      summarise(value = sum(value)) ->
+      L115.EDGAR
+
+    # Scale EPA emission by tech to match EDGAR tools
+    L115.emiss_scaler <- L115.nh3_tg_R_an_C_Yh.mlt
+
+    # First match by GCAM Region ID, gas, EDGAR_agg_sector and year, then compute scalers
+    tidyr::unite(L115.emiss_scaler, col = "Region_GHG_Sector_Yr", c(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year), sep = "~") %>%
+      left_join(tidyr::unite(L115.EDGAR, col = "Region_GHG_Sector_Yr", c(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year), sep = "~"), by = "Region_GHG_Sector_Yr") %>%
+      rename(EDGAR_emissions = value) %>%
+      tidyr::separate(Region_GHG_Sector_Yr, c("GCAM_region_ID", "Non.CO2", "EDGAR_agg_sector", "year"), sep = "~") %>%
+      mutate( ., scaler = EDGAR_emissions / EPA_emissions / 1000.0 ) ->
+      L115.emiss_scaler
+
+    # Now, match by GCAM Region ID, gas, EDGAR_agg_sector and year, then scale EPA emission by tech to match EDGAR tools
+    tidyr::unite(L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt, col = "Region_GHG_Sector_Yr", c(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year), sep = "~") %>%
+      left_join(tidyr::unite(L115.emiss_scaler, col = "Region_GHG_Sector_Yr", c(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year), sep = "~"), by = "Region_GHG_Sector_Yr") %>%
+      tidyr::separate(Region_GHG_Sector_Yr, c("GCAM_region_ID", "Non.CO2", "EDGAR_agg_sector", "year"), sep = "~") %>%
+      mutate(emissions = epa_emissions * scaler) %>%
+      mutate_all(funs(replace(., is.na(.), 0))) ->
+      L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt
+
+    # Rename columns and reshape data
+    L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt %>%
+      rename(supplysector = GCAM_commodity, subsector = system, stub.technology = feed, value = emissions) %>%
```

Comment:
`select` can also be used to rename, FYI; could combine lines 122 and 123 into a single `select` I think.